Makefile

```diff
@@ -0,0 +1,38 @@
+goimports := golang.org/x/tools/cmd/goimports@v0.1.12
+golangci_lint := github.com/golangci/golangci-lint/cmd/golangci-lint@v1.49.0
+
+.PHONY: test
+test:
+	@go test -v ./...
+
+golangci_lint_path := $(shell go env GOPATH)/bin/golangci-lint
+
+$(golangci_lint_path):
+	@go install $(golangci_lint)
+
+.PHONY: lint
+lint: $(golangci_lint_path)
+	@CGO_ENABLED=0 $(golangci_lint_path) run --timeout 5m
+
+.PHONY: format
+format:
+	@find . -type f -name '*.go' | xargs gofmt -s -w
+	@for f in `find . -name '*.go'`; do \
+	    awk '/^import \($$/,/^\)$$/{if($$0=="")next}{print}' $$f > /tmp/fmt; \
+	    mv /tmp/fmt $$f; \
+	done
+	@go run $(goimports) -w -local github.com/http-wasm/http-wasm-host-go `find . -name '*.go'`
+
+.PHONY: check
+check:
+	@$(MAKE) lint
+	@$(MAKE) format
+	@go mod tidy
+	@if [ ! -z "`git status -s`" ]; then \
```

Comment:
I usually dislike this kind of checks. If there are unstaged changes the diff will always fail even if the files are correct. I would first run a `git stash -k -u` and then unstash. See https://github.com/corazawaf/coraza/pull/413

Reply:
Let's say `go mod tidy` made a change. then you do `git stash -k -u`, that stashes the fix tidy did. then `git status -s` passes for the wrong reason.